src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const result = {};
+  const strings = sourceString.split(';').map((str) => str.trim());
+
+  strings.forEach((string) => {
+    const [key, ...values] = string.split(':');
+    const value = values.join('').trim();
+    const trimmedKey = key.trim();
+
+    if (trimmedKey && value) {
+      result[trimmedKey] = value;
```

Comment:
Instead of manually trimming the key and value, you can use a more robust method to handle potential CSS syntax issues, such as multiple colons or extra whitespace within the value.